### DISCUSSION TITLE
Tailwind 4.0 not usable with any system that provides its own styles

### DISCUSSION THREAD

**Discussion by StevenStavrakis** [tailwindlabs/tailwindcss#15466] [Help]
I'm making a plugin for Obsidian. Obsidian comes with its own default set of styles. In previous versions of Tailwind, styles did not compile with layers, allowing the Tailwind styles to override default styles for elements (e.g., h1 tags).

Additionally, Obsidian adds plugin styles by taking the content of the `styles.css` and adding it directly into a `<head>` tag.

Due to the final `styles.css` file retaining the layers, the default global styles of Obsidian overrule tailwind utilities, making it impossible to use utility classes for text color, resizing text, etc. Tailwind 3.x had the `important` flag in its config, which, while I never used it, I assume was partly to address scenarios like this.

I'm missing something, or there is some build-step configuration I can implement to change this behavior, but I haven't found anything so far. This seems a somewhat common issue (see https://github.com/tailwindlabs/tailwindcss/discussions/15074), but the Tailwind team has yet to respond. If, going forward, Tailwind is no longer suitable for projects that involve CSS from other sources, I would prefer to know sooner rather than later.

**Comment by wongjn** [comment]
I asked about this too in #13188.

**Comment by gnat** [comment]
Yes bumped into this too with v4. Creates precedence mess with native `<style>` and existing .css files unless one pro-actively adds `@layer { ... }` everywhere because non-layer elements take selector priority no matter what.

Creates broken situations where tailwind classes on an element no longer apply.

I think we need an option to turn layers off.

**Comment by awacode21** [comment]
i am also running into this problem.. months later.. are there any new findings?

**Reply by wongjn** [reply]
I've been working around it by removing the layer from utilities:

Change your CSS like so:

```diff
-@import "tailwindcss";
+@layer theme, base, components, 
+
+@import "tailwindcss/theme.css" layer(theme);
+@import "tailwindcss/preflight.css" layer(base);
+@import "tailwindcss/utilities.css";
```

**Reply by onlime** [reply]
exactly same solution here, integrating Tailwind v4 into Vitepress (latest v1.6.3) without applying any layers:

```diff
-@import 'tailwindcss';
+@layer theme, base, components, utilities;
+
+@import "tailwindcss/theme.css";
+@import "tailwindcss/utilities.css";
```

**Reply by rozsazoltan** [reply]
https://github.com/tailwindlabs/tailwindcss/discussions/13188#discussioncomment-8737721 --> https://github.com/tailwindlabs/tailwindcss/discussions/13188#discussioncomment-12900719

**Comment by nordkite** [comment]
not sure if this is a good solution but I fixed it by adding utilities into my app wrapper class
```
@layer theme, base, components, utilities;

@import "tailwindcss/theme.css" layer(theme);
@import "tailwindcss/preflight.css" layer(base);

@import "tw-animate-css";
@import "./theme.css";

.my-app {
    @import "tailwindcss/utilities.css";
}
```
so it became `.my-app .flex` for example without `@layer utilities` which prevented applying styles because it has lower priority than the same class on my existing site.

**Reply by rozsazoltan** [reply]
https://github.com/tailwindlabs/tailwindcss/discussions/13188#discussioncomment-8737721 --> https://github.com/tailwindlabs/tailwindcss/discussions/13188#discussioncomment-12900719

Btw, all imports must be at the top of the file. This nested import definitely won't work with native CSS.

https://developer.mozilla.org/en-US/docs/Web/CSS/@import
